### ISSUE TITLE
Terraform Init fails with error `Could not retrieve the list of available versions for provider hashicorp/google`

### ISSUE THREAD

**Issue by shawlz** [terraform-google-modules/terraform-google-project-factory#695]
### TL;DR

Running `terraform init` fails with error `Could not retrieve the list of available versions for provider hashicorp/google` and exit code 1. 

### Expected behavior

Running `terraform init` should succeed with message `Terraform has been successfully initialized!` and exit code 0

### Observed behavior

`Error: Failed to query available provider packages
│ 
│ Could not retrieve the list of available versions for provider
│ hashicorp/google: no available releases match the given constraints >=
│ 2.12.0, >= 3.43.0, >= 3.45.0, >= 3.50.0, >= 3.53.0, ~> 3.53, ~> 4.5, <
│ 5.0.0`

### Terraform Configuration

```hcl
module "project-factory" {
  source  = "terraform-google-modules/project-factory/google"
  version = "12.0.0"
  project = google_project.gke_project.project_id

  activate_apis = [
    "compute.googleapis.com",
    "iam.googleapis.com",
  ]
}
```

### Terraform Version

```sh
Terraform v1.1.7 on darwin_amd64
```

### Additional information

Terraform init and apply works after removing the module. Also works with version = 11.3.1

**Comment by morgante** [comment]
You probably have a version constraint in your configuration that is incompatible with this model. Please check the version constraints defined in your configuration.

This is not a bug in the module.
